spec/models/user_spec.rb

```diff
@@ -360,6 +360,376 @@ def groups_to_impersonate
     end
   end
 
+  describe '#can_admin?' do
+    before :each do
+      expect(subject).not_to receive(:is_manager)
+      expect(subject).not_to receive(:is_viewer)
+    end
+
+    shared_examples 'Argo grants permission to manage object' do
+      it 'permits admin' do
+        expect(subject).to receive(:is_admin).once.and_return(true)
+        expect(subject).not_to receive(:is_manager)
```

Comment:
This is repeated already in the `before` block... Can it be removed from the before block or is it needed in both places?